app/gosqs/send_message.go

```diff
@@ -0,0 +1,133 @@
+package gosqs
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Admiral-Piett/goaws/app/interfaces"
+	"github.com/Admiral-Piett/goaws/app/models"
+
+	"github.com/Admiral-Piett/goaws/app/utils"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/common"
+	"github.com/gorilla/mux"
+)
+
+func SendMessageV1(req *http.Request) (int, interfaces.AbstractResponseBody) {
+	requestBody := models.NewSendMessageRequest()
+	ok := utils.REQUEST_TRANSFORMER(requestBody, req)
+	if !ok {
+		log.Error("Invalid Request - CreateQueueV1")
+		return createErrorResponseV1(ErrInvalidParameterValue.Type)
+	}
+	messageBody := requestBody.MessageBody
+	messageGroupID := requestBody.MessageGroupId
+	messageDeduplicationID := requestBody.MessageDeduplicationId
+	messageAttributes := requestBody.MessageAttributes
+
+	queueUrl := getQueueFromPath(requestBody.QueueUrl, req.URL.String())
+
+	queueName := ""
+	if queueUrl == "" {
+		vars := mux.Vars(req)
+		queueName = vars["queueName"]
```

Comment:
Sorry, I don't know because it was an existing code.
I'm not sure how Go's `http.Request`'s `Context` is set, but all existing tests succeeded even if I removed this mux part.

It may be possible to investigate separately from aws-json support.

Reply:
Awesome, thank you!  I see now it's all over the place...I'll have to think about this more.  But I really appreciate the support!